File: src/main/scala/com/bombproj/dto/ProjectCodeDto.java
package com.bombproj.dto;

import lombok.Data;
import org.hibernate.validator.constraints.Length;

import javax.validation.constraints.NotBlank;

@Data
public class ProjectCodeDto {

    private String id;

    private String projectId;

    @NotBlank(message = "状态码不能为空", groups = AddOrUpdateProjectCodeGroup.class)
    @Length(max = 50, message = "状态码最大长度为 50", groups = AddOrUpdateProjectCodeGroup.class)
    private String code;

    @NotBlank(message = "状态码描述不能为空", groups = AddOrUpdateProjectCodeGroup.class)
    @Length(max = 128, message = "状态码描述最大长度为 128", groups = AddOrUpdateProjectCodeGroup.class)
    private String message;

    private Integer page;

    private String queryText;

    public interface AddOrUpdateProjectCodeGroup {

    }

}

File: src/main/scala/com/bombproj/dto/ProjectCodeDto.scala
package com.bombproj.dto

import javax.validation.constraints.NotBlank
import org.hibernate.validator.constraints.Length

import scala.beans.BeanProperty

case class ProjectCodeDto() {

    @BeanProperty var id: String = _

    @BeanProperty var projectId: String = _

    @BeanProperty
    @NotBlank(message = "状态码不能为空", groups = Array(classOf[AddOrUpdateProjectCodeGroup]))
    @Length(max = 50, message = "状态码最大长度为 50", groups = Array(classOf[AddOrUpdateProjectCodeGroup]))
    var code: String = _

    @BeanProperty
    @NotBlank(message = "状态码描述不能为空", groups = Array(classOf[AddOrUpdateProjectCodeGroup]))
    @Length(max = 128, message = "状态码描述最大长度为 128", groups = Array(classOf[AddOrUpdateProjectCodeGroup]))
    var message: String = _

    @BeanProperty var page: Int = _

    @BeanProperty var queryText: String = _

    trait AddOrUpdateProjectCodeGroup

}
